packages/wonder-blocks-popover/components/popover.test.js

```diff
@@ -0,0 +1,113 @@
+
+describe("Popover", () => {
+    afterEach(() => {
+        unmountAll();
+    });
+
+    it("should set the anchor as the popover ref", () => {
+        // Arrange
+        const wrapper = mount(
+            <Popover
+                placement="top"
+                content={<PopoverContent title="Title" content="content" />}
+            >
+                {({open}) => (
+                    <button data-anchor onClick={open}>
+                        Open default popover
+                    </button>
+                )}
+            </Popover>,
+        );
+
+        // Act
+        const anchorElement = wrapper.find("[data-anchor]").getDOMNode();
+
+        // Assert
+        expect(wrapper.state("anchorElement")).toBe(anchorElement);
+    });
+
+    it("should hide the popover dialog by default", () => {
+        // Arrange, Act
+        const wrapper = mount(
+            <Popover
+                placement="top"
+                content={<PopoverContent title="Title" content="content" />}
+            >
+                {({open}) => (
+                    <button data-anchor onClick={open}>
+                        Open default popover
+                    </button>
+                )}
+            </Popover>,
+        );
+
+        // Assert
+        expect(wrapper.find(PopoverContent)).not.toExist();
+    });
+
+    it("should render a text-only variant", () => {
+        // Arrange
+        const wrapper = mount(
+            <Popover
+                placement="top"
+                content={<PopoverContent title="Title" content="content" />}
+            >
+                {({open}) => (
+                    <button data-anchor onClick={open}>
+                        Open default popover
+                    </button>
+                )}
+            </Popover>,
+        );
+
+        // Act
+        wrapper.find("[data-anchor]").simulate("click");
+
+        // Assert
+        expect(wrapper.find(PopoverContent)).toExist();
+    });
+
+    it("should close the popover from inside the content", () => {
+        // Arrange
+        const onCloseMock = jest.fn();
+
+        const wrapper = mount(
+            <Popover
+                placement="top"
+                onClose={onCloseMock}
+                content={({close}) => (
+                    <PopoverContentCore>
+                        <span>custom popover</span>
+                        <button data-close-button onClick={close}>
+                            close popover
+                        </button>
+                    </PopoverContentCore>
+                )}
+            >
+                {({open}) => (
+                    <button data-anchor onClick={open}>
+                        Open default popover
+                    </button>
+                )}
+            </Popover>,
+        );
+
+        // Act
+        // first open the popover
+        wrapper.find("[data-anchor]").simulate("click");
+        expect(wrapper.find(PopoverContentCore)).toExist();
```

Comment:
These two lines should be part of `Arrange` since the thing we're testing is the closing.  Also, the `expect` isn't necessary since there's another test that checks that popover can be opened from a trigger element.